cpp/tests/utilities/base_fixture.hpp

```diff
@@ -69,7 +69,12 @@ inline auto make_managed() { return std::make_shared<rmm::mr::managed_memory_res
 
 inline auto make_pool()
 {
-  return rmm::mr::make_owning_wrapper<rmm::mr::pool_memory_resource>(make_cuda());
+  auto const [free, total] = rmm::detail::available_device_memory();
+  auto min_alloc =
```

Comment:
Could you document why you want this amount? It's not obvious how this value was selected, or that it's potentially tied to the maximum parallelism used in our test suite.

Also:
```suggestion
  auto const min_alloc =
```

Reply:
Updated the `make_pool` to have a comment on why we have this magic value, and the impact of changing it.